### PR TITLE
User can login to view an article

### PR DESCRIPTION
**Pull request by KyunginNa**
[User can log in and get full access to all articles](https://www.pivotaltracker.com/story/show/176444926)
## User Story
```
As a Registered User,
In order to get access to all articles,
I would like to be able to log in to the mobile app.
```
## Changes proposed in this pull request:
* Adding Stack Screen for SignIn screen
* Adding login button in navigation container 
* Creating SignInScreen component
* Creating auth module

## What I have learned working on this feature:
* How to implement user authentication in react-native
* How to create a button with onPress function

## Packages/Gems added
* expo/vector-icons
* react-native-async-storage/async-storage
* react-native-community/async-storage

## Screenshots (Client)
Screenshot 2021-01-17 at 12 35 29

### CODE REVIEW

File: App.jsx [hunk 1]
```diff
@@ -1,43 +1,66 @@
...
-              color: "rgb(240,230,140)",
-              fontSize: 25,
-              fontWeight: "bold",
+          options={(props) => ({
+            headerRight: () => {
+              return (
+                <FontAwesome5
+                  name="cat"
+                  size={24}
+                  color="white"
```
Comment (ViktorHek) [inline App.jsx:26]: would be nice to change the color of the logo when a user is logged in. but given the limited time, it is understandable if we don't change that.

File: components/DisplayArticlesList.jsx [hunk 2]
```diff
@@ -7,20 +7,23 @@ import {
...
+  const { credentials } = useSelector(state => state)
+
   return (
     <TouchableOpacity
       onPress={() => {
-        navigation.navigate("SingleArticle", { article: article });
+        if (credentials) {
+          navigation.navigate("SingleArticle", { article: article })
+        } else {
+          alert("HISS! Please sign in to read an article!")
```
Comment (ViktorHek) [inline components/DisplayArticlesList.jsx:21]: try open this in the app simulation and see first how alert-message looks and to see if the message is not to long

File: components/SignInScreen.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      />
+      <TextInput
+        secureTextEntry={true}
+        placeholder="Password"
+        onChangeText={(text) => setPassword(text)}
+      />
+      <Button
+        title="Sign-In"
+        color="purple"
+        onPress={() => authenticateUser()}
```
Comment (ViktorHek) [inline components/SignInScreen.jsx]: add padding

File: state/store/initialState.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const initialState = {
-  appHeader: "Kitty News",
+  appHeader: 'Kitty News',
   articles: [],
   currentArticle: {},
-};
+}
```
Comment (ViktorHek) [inline state/store/initialState.js:5]: maybe add `credentials: {}`

**Review by ViktorHek — COMMENTED** [review]
good job

File: App.jsx (hunk 1 above)
Comment (salindholm) [reply App.jsx:26]: We have had that thought, but as a group we decided that time was not enough for this extra feature.

File: components/DisplayArticlesList.jsx (hunk 2 above)
Comment (KyunginNa) [reply components/DisplayArticlesList.jsx:21]: Yea we tried it, and it looked good :)

File: state/store/initialState.js (hunk 4 above)
Comment (KyunginNa) [reply state/store/initialState.js:5]: It works without it, so we decided not to add this one

**Pull request merged**